docs/extensions.adoc

```diff
@@ -328,7 +328,22 @@ class TimedSpec extends Specification {
 When applied to a fixture method, the timeout is per execution of the fixture method.
 
 When a timeout is reported to the user, the stack trace shown reflects the execution stack of the test framework when
-the timeout was exceeded.
+the timeout was exceeded. Additionally, thread dumps can be captured and logged on unsuccessful interrupt attempts if configured using the <<spock-configuration-file,Spock Configuration File>>.
+
+.Timeout Configuration
+[source,groovy]
+----
+timeout {
+  // boolean, default false
+  printThreadDumpsOnInterruptAttempts true
+  // integer, default 3
+  maxInterruptAttemptsWithThreadDumps 5
+  // org.spockframework.runtime.extension.builtin.ThreadDumpUtilityType, default JCMD
+  threadDumpUtilityType threadDumpUtilityType.JSTACK
+  // list of java.lang.Runnable, default []
+  interruptAttemptListeners.add({ println('Timeout occurred!') })
+}
+----
```

Comment:
Ideally, this would be backed by a test, see [here](https://github.com/spockframework/spock/blob/83e304f12fb7b3235474d8cd464af654e3e4b255/docs/parallel_execution.adoc?plain=1#L717-L719).